test/lib/testMayaUsdDirtyScene.py

```diff
@@ -0,0 +1,163 @@
+        with DirtySceneCheck(self):
+            cmds.mayaUsdLayerEditor(layer1Id, edit=True, muteLayer=(True, shapePath))
+        with DirtySceneCheck(self):
+            cmds.mayaUsdLayerEditor(layer1Id, edit=True, muteLayer=(False, shapePath))
+
+        # Remove anon layer -> dirties scene
+        with DirtySceneCheck(self):
+            cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, removeSubPath=(0, shapePath))
+
+        # Reorder anon layer ->dirties scene
+        cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, clear=True)
+        layer1Id = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="Layer1")[0]
+        layer2Id = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="Layer2")[0]
+        layer3Id = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="Layer3")[0]
+        self.assertIsNotNone(layer1Id)
+        self.assertIsNotNone(layer2Id)
+        self.assertIsNotNone(layer3Id)
+        cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, removeSubPath=(2, shapePath))
+        with DirtySceneCheck(self):
+            cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, insertSubPath=(1, layer1Id))
+
+    def testFileBackedLayers(self):
+        mayaUtils.openTopLayerScene()
+        shapePath, stage = getMayaStage()
+
+        # Mute/Un-mute named layer -> dirties scene
+        with DirtySceneCheck(self):
+            stage.MuteLayer('Assembly_room_set.usda')
+        with DirtySceneCheck(self):
+            stage.UnmuteLayer('Assembly_room_set.usda')
+
+        # Remove named layer -> dirties scene
+        rootLayer = stage.GetRootLayer()
+        with DirtySceneCheck(self):
+            cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, removeSubPath=(0, shapePath))
+
+    def testContextOps(self):
+        mayaUtils.openTopLayerScene()
+
+        # Ball_35 has both variants and references.
+        ufePath = ufe.PathString.path('|transform1|proxyShape1,/Room_set/Props/Ball_35')
+        item = ufe.Hierarchy.createItem(ufePath)
+        contextOps = ufe.ContextOps.contextOps(item)
+
+        # Change shading variant -> dirties scene
+        with DirtySceneCheck(self):
+            contextOps.doOp(['Variant Sets', 'shadingVariant', 'Cue'])
+
+        # Unload reference -> dirties scene
+        with DirtySceneCheck(self):
+            contextOps.doOp(['Unload'])
+
+        # Load reference -> dirties scene
+        with DirtySceneCheck(self):
+            contextOps.doOp(['Load'])
+
+    def testDefaultPrim(self):
+        mayaUtils.openTopLayerScene()
+        shapePath, stage = getMayaStage()
+
+        # Change default prim -> dirties scene
+        ufePath = ufe.PathString.path('|transform1|proxyShape1,/Room_set/Props/Ball_35')
+        item = ufe.Hierarchy.createItem(ufePath)
+        raw = item.getRawAddress()
+        prim = mayaUsd.ufe.getPrimFromRawItem(item.getRawAddress())
```

Comment:
If you want to limit the use of getRawAddress(), could use mayaUsd.ufe.ufePathToPrim() with your line 131 path string above.

Reply:
Right, good suggestion.